mlflow/tracking/fluent.py

```diff
@@ -123,17 +111,28 @@ def verify_experiment_active(experiment):
             # simultaneously, a race condition may be encountered here wherein experiment creation
             # fails
             experiment_id = client.create_experiment(experiment_name)
+            experiment = client.get_experiment(experiment_id)
     else:
         experiment = client.get_experiment(experiment_id)
         if experiment is None:
             raise MlflowException(
                 message=f"Experiment with ID '{experiment_id}' does not exist.",
                 error_code=RESOURCE_DOES_NOT_EXIST,
             )
-        verify_experiment_active(experiment)
+
+    if experiment.lifecycle_stage != LifecycleStage.ACTIVE:
+        raise MlflowException(
+            message=(
+                "Cannot set a deleted experiment '%s' as the active experiment."
+                " You can restore the experiment, or permanently delete the "
+                " experiment to create a new one." % experiment.name
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
+        )
 
     global _active_experiment_id
-    _active_experiment_id = experiment_id
+    _active_experiment_id = experiment.experiment_id
+    return experiment
```

Comment:
This actually simplifies some MLOps pipeline flows! A bonus feature is always a great thing